zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+"""Saswell (Tuya whitelabel) 88teujp thermostat valve quirk."""
+
+from functools import reduce
+import logging
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomDevice
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import Basic, Identify, Ota, PowerConfiguration
+from zigpy.zcl.clusters.hvac import Thermostat
+
+from . import TUYA_GET_DATA, TUYA_SET_DATA, TUYA_SET_DATA_RESPONSE, TuyaManufCluster
+from .. import Bus, LocalDataCluster
+from ..const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+OCCUPIED_HEATING_SETPOINT_REPORTED = "occupied_heating_setpoint_reported"
+SYSTEM_MODE_REPORTED = "system_mode_reported"
+LOCAL_TEMP_REPORTED = "local_temp_reported"
+BATTERY_REPORTED = "battery_reported"
+
+OCCUPIED_HEATING_SETPOINT_COMMAND_ID = 615
+SYSTEM_MODE_COMMAND_ID = 357
+LOCAL_TEMP_COMMAND_ID = 614
+BATTERY_STATE_COMMAND_ID = 1385
+
+OCCUPIED_HEATING_SETPOINT_ATTR = 0x0012
+SYSTEM_MODE_ATTR = 0x001C
+RUNNING_MODE_ATTR = 0x001E
+LOCAL_TEMP_ATTR = 0x0000
+CTRL_SEQ_OF_OPER_ATTR = 0x001B
+BATTERY_PERCENTAGE_REMAINING_ATTR = 0x0021
+
+MIN_HEAT_SETPOINT_ATTR = 0x0015
+MAX_HEAT_SETPOINT_ATTR = 0x0016
+
+
+def payload_to_decimal(data):
```

Comment:
The tuya support in `__init__.py` already has some code to handle this, which harnesses the `zigpy.types` serialization support.

Reply:
If you're talking about https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/__init__.py#L61, then the implementation is simply wrong and can't properly parse and serialize values greater that 22 or something degrees(in case of temperatures). My implementation is a rewritten JS code from zigbee2mqtt and it was testsed against all possible values

Reply:
I am very surprised by that assertion. As far as I know it worked for every one so far. 25°C or something would be the 255 value boundary over which two bytes are needed for the value. I am pretty sure the code works (esp. Since I tested it locally with my siren device which has a temperature sensor included)

Reply:
I tested the code again and it works as intended, giving the same results as yours. How did you try to use it ? It expects you to use `zigpy.types.uint32_t` for a 4 byte value as temperatures are in the tuya ecosystem.

Reply:
I probably misread the code. Let's do this: I'll try to reuse everything from `__init__.py`, after that it'll be much easier to incorporate my PR to #538